pkg/controller/release/errors.go

```diff
@@ -220,3 +215,36 @@ func NewDuplicateCapabilityRequirementError(capability string) DuplicateCapabili
 		capability: capability,
 	}
 }
+
+type NotWorkingOnStrategyError error
```

Comment:
This doesn't work as expected. You should not use the `error` interface in there, but a concrete type.

The following program prints `this shouldn't happen` when executed:

```go
package main

import "fmt"

type ErrorA error

func IsErrorA(err error) bool {
	_, ok := err.(ErrorA)
	return ok
}

type ErrorB error

func IsErrorB(err error) bool {
	_, ok := err.(ErrorB)
	return ok
}

func main() {
	var err ErrorA = fmt.Errorf("an error")

	if IsErrorB(err) {
		fmt.Printf("this shouldn't happen")
	}
}
```

In contrast, the following program does the right thing, since the custom type is based on a concrete type, not an interface:

```go
package main

import "fmt"

type ErrorA string

func (e ErrorA) Error() string {
	return string(e)
}

func IsErrorA(err error) bool {
	_, ok := err.(ErrorA)
	return ok
}

type ErrorB string

func (e ErrorB) Error() string {
	return string(e)
}

func IsErrorB(err error) bool {
	_, ok := err.(ErrorB)
	return ok
}

func main() {
	var err ErrorA = "an error"

	if IsErrorB(err) {
		fmt.Printf("this shouldn't happen")
	} else {
		fmt.Printf("this is correct")
	}
}
```

Reply:
I removed the type checker function from errors.go as it was never used. What's left is an error definition and a stringer helper, which is there to communicate the error, not to make any decisions based on the specific type.

Reply:
Removing the checker function is fine, but perhaps we should still change the underlying type to at least a string, and add the Error() method to implement the error interface. If a checker function is required in the future, then it won't work as expected and will be a hidden bug.